src/main/webapp/js/instructorStudentRecords.js

```diff
@@ -22,10 +22,21 @@ $(document).ready(function() {
 
     $('.panel-heading.student_feedback').click(toggleSingleCollapse);
 
-    $('input[type=checkbox]').click(function() {
+    $('input[type=checkbox]').click(function(e) {
         var table = $(this).parent().parent().parent().parent();
         var form = table.parent().parent().parent();
         var visibilityOptions = [];
+        var target = $(e.target);
+        
+        if (target.prop('class').includes('answerCheckbox') && !target.prop('checked')) {
+            target.parent().parent().find('input[class*=giverCheckbox]').prop('checked', false);
```

Comment:
I understand that you have copied this code from the other implementation, let's improve on it by moving away from all the ```.parent()```, namely, please look at this pull request here at https://github.com/TEAMMATES/teammates/pull/5389/files where @JosephineKwa has done a fine example of what I mean, perhaps you could sample that and look at how this whole part can be improved